lib/tty/prompt/list.rb

```diff
@@ -348,10 +469,19 @@ def keyleft(*)
       end
       alias keypage_up keyleft
 
-      def keypress(event)
-        return unless filterable?
+      # Callback fired when a confirm key is pressed
+      #
+      # @api private
+      def confirm
+        @done = true unless choices.empty?
+      end
 
-        if event.value =~ FILTER_KEYS_MATCHER
+      def keypress(event)
+        if !(@confirm_keys.keys & [event.key.name, event.value]).empty?
```

Comment:
You're probably right that this change may have infinitesimal improvement overall. I try to make key pressing methods as snappy as possible though. You never know, this may be used by a super fast typist that filters a very long list and gets a visible flicker in their terminal, then it all will be worth it 😉